research-software-directory/RSD-as-a-service
# 110 license scraper

# License scraper

Changes proposed in this pull request:

* Add a license scraper
* Refactor the existing programming languages scraper
* Licenses are at two places in the database now: in the table `repository_url`, where scraped licenses are stored and in the table `license_for_software`, which contains additional, manually added licenses.

How to test:

* The database structure has changed, so we need to rebuild everything:
* `docker-compose down --volumes`
* `docker-compose build`
* `docker-compose up`
* Run the data migration script
* At 3, 9, 15, etc. minutes past the hour the license scraper should run and print two lines to the console
* Check https://localhost/api/v1/repository_url to see that data is indeed scraped and saved
* (The existing programming languages scraper should run at 0, 6, 12, etc. minutes past the hour)

PR Checklist:


Closes #110
